sunkit_image/nafe.py

```diff
@@ -0,0 +1,411 @@
+
+    nafe_weight : `~float`
+        Typical values for nafe_weight lie in the interval [0.05, 0.3].
+        Controls the ratio of the nafe component in the result image.
+
+    roi_p : `~tuple`
+        (row, col) of ROI top left corner pixel.
+
+    roi_f : `~tuple`
+        (width, height) of ROI frame.
+
+    nproc : `~int`
+        Number of processes for multiprocessing. Defaults to number of
+        available cpu cores.
+
+    hbins : `~int`
+        Number of histogram bins.
+
+    nrs : `~float`
+        Sigma of artificial gaussian noise added to the input image.
+
+    n : `~int`
+        Neighborhood width used as kernel size for nafe transform. Must be an
+        odd integer.
+
+    c : `~list`
+        Fuzzy neighborhood constants. Defaults to 12 layers each of
+        coefficient 1.
+
+    Returns
+    -------
+    enhanced_image : `~numpy.ndarray`
+        The enhanced image resulting from the linear combination of two images:
+        gamma_transform(input_image) and nafe_transform(input_image) such that:
+            enhanced_image = (1 - nafe_weight) * gamma_transform(input_image)
+                             + nafe_weight * nafe_transform(input_image)
+
+    Raises
+    ------
+    ValueError:
+        When neighborhood_width is not an odd integer or gamma is not positive.
+    """
+    if imargin is None:
+        # TODO: implement better default contrast utilizing gb, lb, gw and lw
+        # while taking in consideration roi_p and roi_f
+        imargin = (input_map.data.min(), input_map.data.max())
+    else:
+        input_map.data[input_map.data < imargin[0]] = 0
+        input_map.data[input_map.data > imargin[1]] = 0
+
+    if gamma is None:
+        gamma = _PREDEF_PARAMS.get(input_map.meta["wavelnth"], _PREDEF_PARAMS["def"])[0]
+    elif gamma < 0.0:
+        raise ValueError("gamma should be a non-negative real number! "
+                         "gamma given is:", gamma)
+
+    if omargin is None:
+        omargin = (0, 2 ** 16 - 1)
+
+    if not nafe or nafe_weight is not None and nafe_weight == 0:
+        if nafe_weight is not None and nafe_weight == 0:
+            print("Warning: setting nafe_weight to zero typically disables"
+                  "nafe! nafe_weight given is:", nafe_weight)
+        enhanced_image = _transform(old_margin=imargin, new_margin=omargin, old_value=input_map.data, power=gamma)
+    else:
+        if nafe_weight is None:
+            nafe_weight = _PREDEF_PARAMS.get(input_map.meta["wavelnth"], _PREDEF_PARAMS["def"])[1]
+        elif nafe_weight < 0 or nafe_weight > 1:
+            raise ValueError("nafe_weight should be between 0 and 1. "
+                             "nafe_weight given is:", nafe_weight)
+        elif nafe_weight < 0.05 or nafe_weight > 0.3:
+            print("Warning: typical values for nafe_weight should be"
+                  "between 0.05 and 0.3! nafe_weight given is:",
+                  nafe_weight)
```

Comment:
I thought this would be useful for debugging if the user uses sunkit-image as part of a bigger project and doesn't actually knows nafe_weight beforehand so it helps him trace things.